app/models/discovery_rule.rb

```diff
@@ -36,6 +36,15 @@ class DiscoveryRule < ApplicationRecord
     end
   }
 
+  def unique_priority
+    Taxonomy.enabled_taxonomies.each do |enabled_taxonomy|
+      discovery_rule_ids  =  TaxableTaxonomy.where(taxable_type: 'DiscoveryRule', taxonomy_id: send("#{enabled_taxonomy.singularize}_ids")).pluck(:taxable_id)
+      if DiscoveryRule.unscoped.find(discovery_rule_ids).pluck(:priority).include?(priority)
```

Comment:
Default scope has `with_taxonomy_scope` and some ordering. The taxonomy limits what you can access according to your organization and location. This is actually what we want to use - user in org A should be able to create rule with the same priority as user B. Can you add a test for that actually? Because with `unscoped` this will not be possible.

Looking in the code, I also see that `suggest_priority` does `self.unscoped.maximum(:priority).to_i + STEP` (with unscoped too) which seems to incorrect. You want to have highest priority from your taxonomy, not globally. A fix with a test would be great too :-)

Reply:
Wait, taxonomy is somehow being incorporated the line above, then you want to use `unscoped`.